pretyping/typeclasses.mli

```diff
@@ -133,7 +133,7 @@ val remove_instance_hint : GlobRef.t -> unit
 val solve_all_instances_hook : (env -> evar_map -> evar_filter -> bool -> bool -> bool -> evar_map) Hook.t
 val solve_one_instance_hook : (env -> evar_map -> EConstr.types -> bool -> evar_map * EConstr.constr) Hook.t
 
-val declare_instance : hint_info option -> bool -> GlobRef.t -> unit
+val declare_instance : ?warn:bool -> hint_info option -> bool -> GlobRef.t -> unit
```

Comment:
Why do you want to warn only sometimes?

Reply:
Because this function is called in other situations. E.g., when declaring section variables.

Reply:
I see. I think a comment (or the label name) should hint at when the warning is triggered

Reply:
Just added a comment.